lib/ProkkaAnnotation/Util/ProkkaUtils.py

```diff
@@ -0,0 +1,699 @@
+        if "metagenome" in params and params["metagenome"] == 1:
+            prokka_cmd_list.append("--metagenome")
+        # --rawproduct      Do not clean up /product annotation (default OFF)
+        if "rawproduct" in params and params["rawproduct"] == 1:
+            prokka_cmd_list.append("--rawproduct")
+        # --fast            Fast mode - skip CDS /product searching (default OFF)
+        if "fast" in params and params["fast"] == 1:
+            prokka_cmd_list.append("--fast")
+        # --mincontiglen [N] Minimum contig size [NCBI needs 200] (default "1")
+        if "mincontiglen" in params and params["mincontiglen"]:
+            prokka_cmd_list.extend(["--mincontiglen", str(params["mincontiglen"])])
+        # --evalue [n.n]    Similarity e-value cut-off (default "1e-06")
+        if "evalue" in params and params["evalue"]:
+            prokka_cmd_list.extend(["--evalue", str(params["evalue"])])
+        # --rfam            Enable searching for ncRNAs with Infernal+Rfam (SLOW!) (default "0")
+        if "rfam" in params and params["rfam"] == 1:
+            prokka_cmd_list.append("--rfam")
+        # --norrna          Don"t run rRNA search (default OFF)
+        if "norrna" in params and params["norrna"] == 1:
+            prokka_cmd_list.append("--norrna")
+        # --notrna          Don"t run tRNA search (default OFF)
+        if "notrna" in params and params["notrna"] == 1:
+            prokka_cmd_list.append("--notrna")
+        prokka_cmd_list.append(subject_fasta_filepath)
+        print("Prokka command line: " + str(prokka_cmd_list))
+
+        try:
+            check_output(prokka_cmd_list, cwd=self.scratch)
+        except CalledProcessError as e:
+            pprint(e)
+        return output_dir
+
+    @staticmethod
+    def retrieve_prokka_results(output_dir):
+        """ Gather up the relevant prokka results, load the records from the results files
+
+        :param output_dir:
+        :return: Sequences from the .faa .ffn files and the gff_filepath
+        """
+        faa_file = output_dir + "/mygenome.faa"
+        cds_to_prot = {}
+        for record in SeqIO.parse(faa_file, "fasta"):
+            cds_to_prot[record.id] = str(record.seq)
+        ffn_file = output_dir + "/mygenome.ffn"
+        cds_to_dna = {}
+        for record in SeqIO.parse(ffn_file, "fasta"):
+            cds_to_dna[record.id] = str(record.seq)
+        gff_file = output_dir + "/mygenome.gff"
+        if not os.path.isfile(gff_file):
+            raise ValueError("PROKKA output GFF file is not found")
+
+        prokka_results = namedtuple("prokka_results", "cds_to_prot cds_to_dna gff_filepath")
+        return prokka_results(cds_to_prot, cds_to_dna, gff_file)
+
+    def parse_prokka_results(self, **prokka_parse_parameters):
+        """ Go through the prokka results from the input contigs and then
+        create the features, mrnas and cdss components of the KbaseGenome.Genome object
```

Comment:
Should note here that this is only used for annotating assemblies not genomes

Reply:
ok

Reply:
ok done